concept/type_/mod.rs

```diff
@@ -221,16 +245,43 @@ impl<'a> TypeEdgePropertyEncoding<'a> for Ordering {
     }
 }
 
-pub(crate) trait InterfaceImplementation<'a>:
+pub trait Capability<'a>:
     TypeEdgeEncoding<'a, From = Self::ObjectType, To = Self::InterfaceType> + Sized + Clone + Hash + Eq + 'a
 {
-    type AnnotationType;
+    type AnnotationType: Hash + Eq + Clone + From<Annotation> + Into<Annotation>;
     type ObjectType: TypeAPI<'a>;
     type InterfaceType: KindAPI<'a>;
+    const KIND: CapabilityKind;
 
     fn object(&self) -> Self::ObjectType;
 
     fn interface(&self) -> Self::InterfaceType;
+
+    fn get_annotations_declared<'this>(
```

Comment:
I don't remember if I **had** to move annotations here alongside cardinality or if I just wanted to move `get annotations` and `get cardinality` together. However, I didn't want to move `set`ters here. Not sure if it's bad though...